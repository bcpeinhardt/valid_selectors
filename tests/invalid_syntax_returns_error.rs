use valid_selectors::xpath;

fn main() {
    let all_links: &str = xpath!("//a/[invalid syntax]");
    assert_eq!(all_links, "//a/[invaid syntax]");
}