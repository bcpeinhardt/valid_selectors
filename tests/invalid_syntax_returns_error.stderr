XPath error : Invalid expression
//a/[invalid syntax]
    ^
error: macro expansion ignores token `;` and any following
 --> tests/invalid_syntax_returns_error.rs:4:27
  |
4 |     let all_links: &str = xpath!("//a/[invalid syntax]");
  |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ caused by the macro expansion here
  |
  = note: the usage of `xpath!` is likely invalid in expression context

error: Invalid XPath
 --> tests/invalid_syntax_returns_error.rs:4:27
  |
4 |     let all_links: &str = xpath!("//a/[invalid syntax]");
  |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `xpath` (in Nightly builds, run with -Z macro-backtrace for more info)
